app/src/viewmodels/map/MapScreenViewModel.tsx

```diff
@@ -26,21 +21,23 @@ const defaultLocation = {
 export function useMapScreenViewModel(
   firestoreCtrl: FirestoreCtrl,
   navigation: any,
+  firstLocation: GeoPoint | undefined,
 ) {
   const [permission, setPermission] = useState<boolean>(false);
-  const [userLocation, setUserLocation] = useState<LocationObject | undefined>(
-    undefined,
+  const [userLocation, setUserLocation] = useState<GeoPoint | undefined>(
+    firstLocation,
```

Comment:
What is 'firstLocation' ? Is it default location ?

Reply:
Well that's quite a tricky vocabulary issue : the `defaultLocation` variable is the location that is used when no location is given in the parameters of the screen AND the user refuses to give their location. The `firstLocation` is the location that *may* be given through the screen parameters. I agree the naming is not very clear, do you have any suggestions ? There are already comments explaining this above the variable declaration.

Reply:
OK I see ! It's fine then